docs/source/cli_help/fork-point.rst

```diff
@@ -0,0 +1,65 @@
+.. role:: bash(code)
+
+.. _fork-point:
+
+fork-point
+----------
+Usage:
+
+.. code-block:: shell
+
+  git machete fork-point [--inferred] [<branch>]
+  git machete fork-point --override-to=<revision>|--override-to-inferred|--override-to-parent [<branch>]
+  git machete fork-point --unset-override [<branch>]
+
+Note: in all three forms, if no <branch> is specified, the currently checked out branch is assumed.
+The branch in question does not need to occur in the definition file.
+
+
+Without any option, displays full SHA of the fork point commit for the <branch>..
+Fork point of the given <branch> is the commit at which the history of the <branch> diverges from history of any other branch.
+
+Fork point is assumed by many :bash:`git machete` commands as the place where the unique history of the <branch> starts.
+The range of commits between the fork point and the tip of the given branch is, for instance:
+    * listed for each branch by :bash:`git machete status --list-commits`
+    * passed to :bash:`git rebase` by :bash:`git machete reapply/slide-out/traverse/update`
+    * provided to :bash:`git diff/log` by :bash:`git machete diff/log`.
+
+:bash:`git machete` assumes fork point of <branch> is the most recent commit in the log of <branch> that has NOT been introduced on that very branch,
+but instead occurs on a reflog (see help for :bash:`git reflog`) of some other, usually chronologically earlier, branch.
+This yields a correct result in typical cases, but there are some situations
+(esp. when some local branches have been deleted) where the fork point might not be determined correctly.
+Thus, all rebase-involving operations (:bash:`reapply`, :bash:`slide-out`, :bash:`traverse` and :bash:`update`) run :bash:`git rebase` in the interactive mode,
+unless told explicitly not to do so by :bash:`--no-interactive-rebase` flag, so that the suggested commit range can be inspected before the rebase commences.
+Also, :bash:`reapply`, :bash:`slide-out`, :bash:`squash`, and :bash:`update` allow to specify the fork point explicitly by a command-line option.
+
+:bash:`git machete fork-point` is different (and more powerful) than :bash:`git merge-base --fork-point`,
+since the latter takes into account only the reflog of the one provided upstream branch,
+while the former scans reflogs of all local branches and their remote tracking branches.
+This makes git machete's :bash:`fork-point` more resilient to modifications of the tree definition which change the upstreams of branches.
```

Comment:
```suggestion
This makes git machete's :bash:`fork-point` more resilient to modifications of .git/machete file where upstreams (parent) of certain branches are changed.
```

Reply:
(for more clarity, the concept of fork-point & its consequences are already convoluted as f**k, that's literally my least favorite aspect of this entire tool xD)

Reply:
(esp. yellow edges are a cancer to explain & handle)